_posts/2017-11-10-choosing-a-cni-provider.md

```diff
@@ -0,0 +1,315 @@
+custom solution for pod CIDR allocation to the nodes. Kube-router also uses
+standard CNI plug-ins so does require any additional CNI plug-in. Kube-router is
+built on standard Linux networking toolset and technologies like ipset,
+iptables, ipvs, and lvs.
+
+Murali Reddy founder of kube-router.
+
+### romana
+
+Chris Marino summarized romana.
+
+Romana uses standard layer 3 networking for pod networks. Romana supports
+Kubernetes Network Policy APIs and never uses an overlay, even when a cluster is
+split across network availability zones.  Romana is the only CNI provider that
+uses native VPC networking across availability zones for HA clusters, delivering
+a high-performance CNI solution. The current release uses its Etcd cluster, but
+the next version will optionally allow the Kubernetes Etcd cluster to be used as
+a datastore.
+
+### weave
+
+Paul Fremantle's synopsis of weave.
+
+Weave supports an overlay network that can span different cloud networking
+configurations, simplifying running legacy workloads on Kubernetes. For example,
+Weave supports multicast, even when the underlying network doesn't. Weave can
+configure the underlying VPC networking and bypass the overlay when running on
+AWS.  This provider forms a mesh network of hosts that are partitionable and
+eventually consistent, meaning that the setup is almost zero-config, and it
+doesn't need to rely on an Etcd. Weave supports encryption and Kubernetes
+network policy ensuring that there is security at the network level.
+
+## By the Numbers
+
+kops does not track usage numbers of the different CNI providers, and I hope
+never does.  When making a product selection, of software hosted on GitHub, I
+look at three different numbers.
+
+1. GitHub Stars - Likes on Github
+1. GitHub Forks - Number of copies of the repo
+1. GitHub Contributors - Number of people with merged code
+
+Github Stars are akin to likes on a Social Media platform.  The number of
+contributors talks to the number of people maintaining the code base and
+documentation.  Active projects have a high number of contributors. The number
+of forks is a mix of likes and contributors. Contributors typically have to fork
+the repo. Other people will fork the project to build a custom copy, push code
+to a feature branch that they own, or for various reasons. The activity level of
+a project is critical.  These are some of the metrics that I use to judge the
+activity level.
+
+![Project Stars](/img/cni-github-03.png){:class="img-responsive"}
+
+![Project Stars](/img/cni-github-02.png){:class="img-responsive"}
+
+![Project Stars](/img/cni-github-01.png){:class="img-responsive"}
+
+## Support Matix
+
+Here is a table of different features of each of the CNI providers mentioned.
+
+<table class="custom-table">
+  <thead>
+    <tr>
+      <th>Provider</th>
+      <th>Network <br>Model</th>
+      <th>BGP</th>
+      <th>Network <br>Policies</th>
+      <th>Mesh</th>
+      <th>External <br>Database</th>
```

Comment:
This one's tricky, because this a negative, right?  I guess I would put "Datastore" and have 3 options "Kubernetes", "Mesh", or "BYO etcd".